pkg/hubble/filters/reply.go

```diff
@@ -29,30 +27,11 @@ func filterByReplyField(replyParams []bool) FilterFunc {
 		}
 		switch f := ev.Event.(type) {
 		case v1.Flow:
-			// FIMXE: In events originating from TraceNotify, the `reply`
-			// field is determined from the connection tracking state in
-			// the datapath. Unfortunately, not all trace points have the
-			// connection tracking state available. For certain trace point
-			// events, we do not know if it actually was a reply or not.
-			//
-			// Ideally, we would fix this in the parser and make the `reply`
-			// an optional boolean, so we can distinguish between a `false`
-			// value and an absent value. This however is a breaking change
-			// in the API. Therefore, we only report flows here for which we
-			// know that the reply field is reliable:
-			if f.GetEventType().GetType() == monitorAPI.MessageTypeTrace {
-				obsPoint := uint8(f.GetEventType().GetSubType())
-				if !monitorAPI.TraceObservationPointHasConnState(obsPoint) {
-					return false
-				}
+			if f.GetIsReply() == nil {
+				return false
 			}
-			// For PolicyVerdict and Drop events, we statically assume they
-			// always have `reply=false`, as the flow is not yet established
-			// when the policy verdict is taken.
-			// For all other events (including Accesslog/L7), we assume the
-			// parser populates the Reply field reliably.
```

Comment:
Reviewer note: This assumption is not presumed anymore with this change, so there is a slight change in behavior here: The reply filter will not report any Drop or Policy Verdict events anymore (as they have `is_reply` always nil).

If people feel strongly about this, I'm happy to add an exception in the filter logic here to keep this previous assumption here. But I'm not 100% confident that it is always correct.